crates/matrix-sdk-ui/src/timeline/event_item/local.rs

```diff
@@ -84,3 +87,29 @@ pub enum EventSendState {
         event_id: OwnedEventId,
     },
 }
+
+/// This type represents the "send progress" of a local event timeline item.
+#[derive(Clone, Debug)]
+pub enum EventSendProgress {
+    /// A media is being uploaded.
+    MediaUploadProgress {
+        /// The index of the media within the transaction. A file and its
+        /// thumbnail share the same index.
+        index: u64,
+
+        /// Is the media a thumbnail?
+        is_thumbnail: bool,
+
+        /// The current upload progress.
+        progress: TransmissionProgress,
+    },
+
+    /// A media has finished uploading.
+    UploadedMedia {
+        /// The index of the media within the transaction. A file and its
+        /// thumbnail share the same index.
+        index: u64,
+        /// Is the media a thumbnail?
+        is_thumbnail: bool,
+    },
```

Comment:
This feels weird. Why is this a separate variant, rather than a `progress` of 0% for the next `index`, or a progress of 100% if it's the last file being uploaded?

Reply:
These variants are created from `RoomSendQueueUpdate` variants and `RoomSendQueueUpdate::UploadedMedia` already existed, so I tried not to break with it. I suppose technically it wouldn't be needed anymore though because, as you say, `MediaUploadProgress` with `current == total` means it's uploaded.

Reply:
Right, I'm not familiar with the send queue API. Let's wait for feedback from the team :)

Reply:
It is definitely a fair question though!